### PR TITLE
Fixed Makefile for MacOS ARM 64 Go bindings

### PR DESCRIPTION
**Pull request by gleicon**
Fixed Makefile for MacOS ARM 64 based on https://github.com/ggerganov/whisper.cpp/issues/1344 + proper ggml-metal env var setting. 
Originally it couldn't find includes and the GGML-Metal files as referenced in the issue above and in the main issue for Go bindings https://github.com/ggerganov/whisper.cpp/discussions/312. Hope that helps.

### CODE REVIEW

**Comment by ggerganov** [comment]
Looks like this change fixes it for MacOS, but it breaks it for non-MacOS

**Comment by gleicon** [comment]
Fixing it, long time I don't build for linux like this and haven't noticed that -framework works only for MacOS on LLVM.

File: bindings/go/Makefile [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
 whisper: mkdir
 	@echo Build whisper
 	@${MAKE} -C ../.. libwhisper.a
 
 test: model-small whisper modtidy
-	@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} go test -v .
-	@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} go test -v ./pkg/whisper/...
+ifeq ($(UNAME_S),Darwin)
+		@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} GGML_METAL_PATH_RESOURCES=${GGML_METAL_PATH_RESOURCES} go test -ldflags "-extldflags '$(EXT_LDFLAGS)'" -v .
+       		@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} GGML_METAL_PATH_RESOURCES=${GGML_METAL_PATH_RESOURCES} go test -ldflags "-extldflags '$(EXT_LDFLAGS)'" -v ./pkg/whisper/...
```
Comment (ggerganov) [inline bindings/go/Makefile:34]:
```suggestion
		@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} GGML_METAL_PATH_RESOURCES=${GGML_METAL_PATH_RESOURCES} go test -ldflags "-extldflags '$(EXT_LDFLAGS)'" -v .
		@C_INCLUDE_PATH=${INCLUDE_PATH} LIBRARY_PATH=${LIBRARY_PATH} GGML_METAL_PATH_RESOURCES=${GGML_METAL_PATH_RESOURCES} go test -ldflags "-extldflags '$(EXT_LDFLAGS)'" -v ./pkg/whisper/...
```

**Review by ggerganov — APPROVED** [review]
Will wait for CI to pass and merge

**Comment by gleicon** [comment]
Thanks !

**Pull request merged**